### PR TITLE
[New] Add openshare w/ git auto-update

### LINKED ISSUE #9639: [Request] Add Openshare
**Library name: Openshare** 
**Git repository url: https://github.com/OpenShare/openshare**
**npm package url(optional): https://www.npmjs.com/package/openshare** 
**License(s): MIT**
**Official homepage: https://openshare.social/**
**Wanna say something? Leave message here:**

=====================
Notes from cdnjs maintainer:
Please read the README.md and CONTRIBUTING.md document first.

You are welcome to add a library via sending pull request,
it'll be faster than just opening a request issue,
and please don't forget to read the guidelines for contributing, thanks!!

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/39290130-request-add-openshare?utm_campaign=plugin&utm_content=tracker%2F32893&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F32893&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by sashberd**
close #9639, cc @OpenShare, using web-minify-helper for minifing files

Pull request for issue: #9639
Related issue(s): # #

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/OpenShare/openshare
 * Official website (optional, not the repository): http://openshare.social/
 * NPM package url (optional): https://www.npmjs.com/package/openshare
 * License and its reference: MIT
 * GitHub / Bitbucket popularity (required):
   - Count of watchers: 11
   - Count of stars: 101
   - Count of forks: 15
 * NPM download stats (optional):
   - Downloads in the last day: 0
   - Downloads in the last week: 16
   - Downloads in the last month: 174

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [ ] The lib has notable popularity
   * [ ] More than 200 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [ ] More than 800 downloads stats per month on npm registry
 * [x] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [ ] Auto-update setup
 * [ ] Auto-update target/source is valid.
 * [ ] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

**Comment by mercyg** [comment]
@PeterDaveHello is this library popular enough to be added?

**Comment by PeterDaveHello** [comment]
@mercyg not really, looks like its current popularity meets our old requirement back to Nov 2016, as @sashberd sent the PR already, maybe we can accept it.

**Comment by sashberd** [comment]
@metcyg @PetetDaveHello
I made  PR's only  for requests that were approoved by our team. However, you can reject and close any of them... its Ok... I will not be offended☺

**Comment by PeterDaveHello** [comment]
@sashberd hmmm I'm not sure if it's really approved, though I don't mind this one, but I think we should work on high priority issues first, thanks.

File: ajax/libs/openshare/package.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    "name": "Digital Surgeons"
+  },
+  "autoupdate": {
+    "source": "git",
+    "target": "git://github.com/OpenShare/openshare.git",
+    "fileMap": [
+      {
+        "basePath": "dist",
+        "files": [
+          "**/openshare*"
```
Comment (sufuf3) [inline ajax/libs/openshare/package.json]: Let's use `"**/!(test)*"`
Many thanks.

File: ajax/libs/openshare/package.json [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    "url": "git+https://github.com/OpenShare/openshare.git"
+  },
+  "keywords": [
+    "social",
+    "share",
+    "sharing"
+  ],
+  "author": {
+    "name": "Digital Surgeons"
+  },
```
Comment (sufuf3) [inline ajax/libs/openshare/package.json]: Let's use one line.
`"author": "Digital Surgeons"`
https://github.com/OpenShare/openshare/blob/master/package.json#L42
Many thanks.

**Comment by sufuf3** [comment]
The commit message, Let's cc author(real person) not organization `cc @OpenShare`
Thanks.

**Comment by sashberd** [comment]
@sufuf3 The PR commit message was updated. FYR

**Comment by sashberd** [comment]
@sufuf3 ping

**Comment by sufuf3** [comment]
@sashberd Would you please try to modify this PR first? This PR is created by you.

**Comment by sashberd** [comment]
@sufuf3 The PR was updated. FYR

**Comment by sashberd** [comment]
@sufuf3 ping

**Comment by sashberd** [comment]
@sufuf3 @extend1994 ping

**Comment by sashberd** [comment]
@xtuc done

File: ajax/libs/openshare/package.json [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    "target": "git://github.com/OpenShare/openshare.git",
+    "fileMap": [
+      {
+        "basePath": "dist",
+        "files": [
+          "**/!(test)*"
+        ]
+      }
+    ]
+  },
```
Comment (MattIPv4) [inline ajax/libs/openshare/package.json]: Can we use npm auto-update instead, please?

https://www.npmjs.com/package/openshare

```
tree node_modules/openshare

node_modules/openshare
├── README.md
├── analytics.js
├── async-test.html
├── bower.json
├── count.js
├── dist
│   ├── openshare.js
│   ├── openshare.min.js
│   └── test.js
├── lib
│   ├── countReduce.js
│   ├── dashToCamel.js
│   ├── init.js
│   ├── initializeCountNode.js
│   ├── initializeNodes.js
│   ├── initializeShareNode.js
│   ├── initializeWatcher.js
│   ├── setData.js
│   ├── share.js
│   └── storeCount.js
├── package.json
├── share.js
├── src
│   ├── browser.js
│   ├── index.js
│   ├── modules
│   │   ├── count-api.js
│   │   ├── count-transforms.js
│   │   ├── count.js
│   │   ├── data-attr.js
│   │   ├── events.js
│   │   ├── open-share.js
│   │   ├── share-api.js
│   │   └── share-transforms.js
│   └── test.js
└── test.html

4 directories, 32 files
```

**Pull request merged**